### PR TITLE
Add ability to parse code

### PR DESCRIPTION
**Pull request by santib**
Something like this is needed so we can load codebases in Vector DBs

### CODE REVIEW

**Comment by andreibondarev** [comment]
@santib This opens up an interesting problem -- how do we chunk code?

**Comment by santib** [comment]
> @santib This opens up an interesting problem -- how do we chunk code?

Yeah, I just used the Text one for simplicity, but checking https://python.langchain.com/docs/modules/data_connection/document_transformers/text_splitters/code_splitter seems like they [get the separators](https://github.com/langchain-ai/langchain/blob/9b24f0b067d9f4a5f3e1f53fe3f7342f79a1f010/libs/langchain/langchain/text_splitter.py#L697) for each language, and that's it.

I can change this PR to do something similar if you want

**Comment by andreibondarev** [comment]
@santib Yeah, I like that!

File: lib/langchain/data.rb [hunk 1]
```diff
@@ -3,26 +3,31 @@
...
 
-    # @param opts [Hash] options passed to the chunker
+    # @param options [Hash] options passed to the chunker
     # @return [Array<String>]
-    def chunks(opts = {})
-      Langchain::Chunker::Text.new(@data, **opts).chunks
+    def chunks(options = {})
+      if Langchain::Processors::Code::EXTENSIONS.include?(source_type)
+        options = options.merge(separators: Langchain::Chunker::Base::LANG_SEPARATORS[source_type])
+        Langchain::Chunker::RecursiveText.new(@data, **options).chunks
```
Comment (santib) [inline lib/langchain/data.rb:27]: TODO: add tests

**Comment by andreibondarev** [comment]
Closing due to inactivity.

**Pull request closed without merging**